tests/providers/amazon/aws/hooks/test_datasync.py

```diff
@@ -50,21 +49,13 @@ def test_get_conn(self):
 
 @mock_datasync
 @mock.patch.object(DataSyncHook, "get_conn")
-class TestDataSyncHookMocked(unittest.TestCase):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.source_server_hostname = "host"
-        self.source_subdirectory = "somewhere"
-        self.destination_bucket_name = "my_bucket"
-        self.destination_bucket_dir = "dir"
+class TestDataSyncHookMocked:
+    source_server_hostname = "host"
+    source_subdirectory = "somewhere"
+    destination_bucket_name = "my_bucket"
+    destination_bucket_dir = "dir"
 
-        self.client = None
-        self.hook = None
-        self.source_location_arn = None
-        self.destination_location_arn = None
-        self.task_arn = None
-
-    def setUp(self):
+    def setup_method(self, method):
```

Comment:
Original @vandonr-amz 
> Ah yes I imagine because of the order of arguments, but I imagine in test_datasync it's not needed ?

Actually this arg needed because we decorated entire class 

```
test setup failed
args = (<tests.providers.amazon.aws.hooks.test_datasync.TestDataSyncHookMocked object at 0x109fdf4c0>, <bound method TestData...HookMocked.test_init of <tests.providers.amazon.aws.hooks.test_datasync.TestDataSyncHookMocked object at 0x109fdf4c0>>)
kwargs = {}

    def wrapper(*args, **kwargs):
        self.start(reset=reset)
        try:
>           result = func(*args, **kwargs)
E           TypeError: setup_method() takes 1 positional argument but 2 were given
```